### ISSUE TITLE
Add getBlockTime RPC API

### ISSUE THREAD

**Issue by mvines** [solana-labs/solana#7115]
It's strongly desired by some users to have a network-controlled best effort timestamp associated with the production time of each block.   One such example is https://en.bitcoin.it/wiki/Block_timestamp

Some important properties of this timestamp are:
1) Drift of no worse than a couple hours from the actual block production time
2) Replaying the ledger at a later time results in the same block production time
2) The block production time is not controlled by a single centralized oracle, but ideally is a function that uses inputs from all validators

A rough proposal would be to require validators to submit their current UTC time every N minutes.   A median time is then constructed from the latest submission and block times are then computed as `last_median_time + slot_offset * 400ms` until the next median time is constructed.

This issue can be broken up into:
1. The definition/implementation of the `getBlockTime` RPC API
2. Initial stub implementation where `last_median_time` is fixed to 0.
3. Adding time submission to all validators and producing a real `last_median_time`

1 and 2 are important to implement quickly to unblock some users

**Comment by CriesofCarrots** [comment]
Remaining steps after #7233 :
- [ ] Calculate stake-weighted mean timestamp (https://docs.solana.com/book/v/master/proposals/validator-timestamp-oracle#calculating-stake-weighted-mean-timestamp)
- [ ] Calculate estimated block time, and return real data in getBlockTime rpc
- [ ] Incorporate timestamp expectations in validator uptime calculations

**Comment by CriesofCarrots** [comment]
Will create a new issue for item #3 above
